packages/components/src/internal/renderers/DataClassTemplateDownloadRenderer.tsx

```diff
@@ -31,7 +33,30 @@ export class DataClassTemplateDownloadRenderer extends React.PureComponent<Props
             });
     };
 
+    fetchTemplates = async (): Promise<ImportTemplate[]> => {
+        const { row } = this.props;
+        return new Promise((resolve, reject) => {
+            getQueryDetails({
+                schemaName: SCHEMAS.DATA_CLASSES.SCHEMA,
+                queryName: row.getIn(['Name', 'value']) ?? row.getIn(['name', 'value']),
+            })
+                .then(queryInfo => {
+                    resolve(queryInfo.getCustomTemplates());
+                })
+                .catch(error => {
+                    reject(error);
+                });
+        });
+    };
+
     render(): ReactNode {
-        return <TemplateDownloadButton onClick={this.onDownload} text="Download" className="button-small-padding" />;
+        return (
+            <TemplateDownloadButton
+                getExtraTemplates={this.fetchTemplates}
```

Comment:
as discussed, this prop will get removed and the schema/query prop will always be passed in

Reply:
done